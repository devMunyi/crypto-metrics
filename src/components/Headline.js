import React from 'react';
import PropTypes from 'prop-types';
import cryptoImg from './cryptos.png';

function Headline({ page = '', cryptoDetails = {}, cryptoCount = 0 }) {
  return (
    <div
      className="row border headline"
      style={{ boxShadow: 'rgba(0, 0, 0, 0.24) 0px 3px 8px' }}
    >
      <div className="col img-container">
        <img
          src={cryptoImg}
          alt="CRYPTO"
          className="img-fluid"
        />
      </div>
      <div className="col d-flex justify-content-center align-items-center border-start">
        <div className="div">
          <h4>{page === 'list' ? 'CRYPTO METRICS' : cryptoDetails.name}</h4>
          <p className="fw-normal">
            {' '}
            {page === 'list'
              ? `${cryptoCount} cryptocurrencies`
              : `${cryptoDetails.priceUsd} usd`}
          </p>
        </div>
      </div>
    </div>
  );
}

Headline.propTypes = {
  page: PropTypes.string.isRequired,
  cryptoDetails: PropTypes.objectOf(PropTypes.string).isRequired,
  // eslint-disable-next-line comma-dangle
  cryptoCount: PropTypes.number.isRequired,
};

export default Headline;
