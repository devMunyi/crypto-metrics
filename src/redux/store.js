import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import cryptoReducer from './reducers';

export default configureStore(
  {
    reducer: {
      crypto: cryptoReducer,
    },
  },
  applyMiddleware(logger),
);
