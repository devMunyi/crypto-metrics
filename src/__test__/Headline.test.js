import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Headline from '../components/Headline';

/* eslint-disable comma-dangle */
it('renders Headline Component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Headline cryptoCount={0} cryptoDetails={{}} page="" />
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});
