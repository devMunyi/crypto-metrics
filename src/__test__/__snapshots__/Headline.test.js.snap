// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Headline Component correctly 1`] = `
<div
  className="row border headline"
  style={
    Object {
      "boxShadow": "rgba(0, 0, 0, 0.24) 0px 3px 8px",
    }
  }
>
  <div
    className="col"
    style={
      Object {
        "height": "200px",
      }
    }
  >
    <img
      alt="CRYPTO"
      className="img-fluid"
      src="cryptos.png"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="col d-flex justify-content-center align-items-center border-start"
  >
    <div
      className="div"
    >
      <h4 />
      <p
        className="fw-normal"
      >
         
        undefined usd
      </p>
    </div>
  </div>
</div>
`;
